GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -54,7 +58,7 @@ public partial class GitIndexProjection : IDisposable, IProfilerOnlyIndexProject
 
         // nonDefaultFileModes is only populated when the platform supports file mode
         // On platforms that support file modes, file paths that are not in nonDefaultFileModes have mode 644
-        private Dictionary<string, ushort> nonDefaultFileModes = new Dictionary<string, ushort>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, ushort> nonDefaultFileTypesAndModes = new Dictionary<string, ushort>(StringComparer.OrdinalIgnoreCase);

```

Comment:
Since this is a sparse data structure anyway, could we just store the result as a class with pre-parsed fields? I saw that we're already masking out the sub-fields when parsing anyway, then storing the combined ushort, then the Mac virtualizer has to mask them out again.

Reply:
>could we just store the result as a class with pre-parsed fields? 

Yep we could do that.  

Do you have a strong preference on storing the data as separate fields vs a single `ushort` with properties that mask out the values for the caller?